cpp/tools/parquet/parquet_dump_footer.cc

```diff
@@ -0,0 +1,246 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdint>
+#include <cstring>
+#include <exception>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <optional>
+#include <random>
+
+#include <thrift/protocol/TCompactProtocol.h>
+#include <thrift/transport/TBufferTransports.h>
+
+#include "arrow/filesystem/filesystem.h"
+#include "arrow/util/endian.h"
+#include "parquet/thrift_internal.h"
+#include "generated/parquet_types.h"
+
+using apache::thrift::protocol::TCompactProtocol;
+using apache::thrift::transport::TMemoryBuffer;
+using apache::thrift::transport::TTransport;
+
+namespace {
+int PrintHelp() {
+  std::cerr << R"(
+Usage: parquet-dump-footer
+  -h|--help    Print help and exit
+  --no-scrub   Do not scrub potentially PII metadata
+  --json       Output JSON instead of binary
+  --in         Input file: required
+  --out        Output file: defaults to stdout
+
+  Dumps the footer of a Parquet file to stdout or a file, optionally with
+  potentially PII metadata scrubbed.
+)";
+  return 1;
+}
+
+uint32_t ReadLE32(const void* p) {
+  uint32_t x;
+  memcpy(&x, p, sizeof(x));
+  return arrow::bit_util::FromLittleEndian(x);
+}
+
+void AppendLE32(uint32_t v, std::string* out) {
+  v = arrow::bit_util::ToLittleEndian(v);
+  out->append(reinterpret_cast<const char*>(&v), sizeof(v));
+}
+
+template <typename T>
+bool Deserialize(const char* data, uint32_t len, T* obj) {
+  parquet::ThriftDeserializer des(/*string_size_limit=*/10 << 20,
+                                  /*container_size_limit=*/10 << 20);
+  try {
+    des.DeserializeMessage(reinterpret_cast<const uint8_t*>(data), &len, obj);
+    return true;
+  } catch (const std::exception& e) {
+    std::cerr << "Failed to deserialize: " << e.what() << "\n";
+    return false;
+  }
+}
+
+template <typename T>
+bool Serialize(const T& obj, std::string* out) {
+  parquet::ThriftSerializer ser(/*initial_buffer_size=*/10 << 20);
+  try {
+    ser.SerializeToString(&obj, out);
+    return true;
+  } catch (const std::exception& e) {
+    std::cerr << "Failed to serialize: " << e.what() << "\n";
+    return false;
+  }
+}
+
+// Replace the contents of s with random data of the same length.
+void Scrub(std::string* s) {
+  static char pool[4096];
+  static std::mt19937 rng(std::random_device {}());
+  static const bool kPoolInit = [] {
+    std::uniform_int_distribution<> caps(65, 90);
+    for (size_t i = 0; i < sizeof(pool); i++) pool[i] = caps(rng);
+    return true;
+  }();
+  (void)kPoolInit;
+
+  const size_t n = s->size();
+  s->clear();
+  while (s->size() < n) {
+    size_t m = std::min(n, sizeof(pool) / 2);
+    std::uniform_int_distribution<> start(0, sizeof(pool) / 2);
```

Comment:
not clear on why we aren't using the full pool here?

Reply:
Added comment and rewritten slightly to make it more random.